parl/common/replay_buffer.py

```diff
@@ -0,0 +1,132 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import random
+from parl.common.error_handling import LastElementError, check_error
+
+
+class Experience(object):
+    def __init__(self, sensor_inputs, states, actions, game_status):
+        check_error('TypeError', list, type(sensor_inputs))
+        self.sensor_inputs = sensor_inputs  # (observation, reward)
```

Comment:
We need to have a discussion about how to name this variable. For a complex case, the inputs might not be sensor data. They might include some intermediate inputs computed by other algorithms. So I suggest using self.inputs (or another better name) throughout all the code.